pkg/controller/authentication/deployment.go

```diff
@@ -305,11 +306,21 @@ func generateDeploymentObject(instance *operatorv1alpha1.Authentication, scheme
 	ldapCACert := instance.Spec.AuthService.LdapsCACert
 	routerCertSecret := instance.Spec.AuthService.RouterCertSecret
 
+	metaLabels := pkgCommon.MergeMap(map[string]string{"app": deployment}, instance.Spec.Labels)
+	podMetadataLabels := map[string]string{
+		"app":                        deployment,
+		"k8s-app":                    deployment,
+		"component":                  deployment,
+		"app.kubernetes.io/instance": "platform-auth-service",
+		"intent":                     "projected",
+	}
+	podLabels := pkgCommon.MergeMap(podMetadataLabels, instance.Spec.Labels)
```

Comment:
If the intent is for the `podMetadataLabels` to be preserved, then the arguments should be reordered.

Reply:
Hi @rwhundley , instead of swapping the arguments in this function, In the MergeMap function body i have swapped it, meaning that i have considered mergeMap as a source(instance.Spec.Labels) to add back podMetadataLables to mergeMap, so that we preserve the podMetadataLabels